packages/server/src/create-blitz-app.ts

```diff
@@ -0,0 +1,63 @@
+import next from "next"
+import { build } from "./build"
+import { alreadyBuilt } from "./build-hash"
+import { normalize, ServerConfig } from "./config"
+import { configureStages } from "./stages"
+
+const debug = require("debug")("blitz:create-blitz-app")
+interface CreateBlitzAppConfig {
+  dev: boolean
+}
+
+export async function createBlitzApp({ dev }: CreateBlitzAppConfig) {
+  const serverConfig: ServerConfig = {
+    env: dev ? "dev" : "prod",
+    rootFolder: process.cwd(),
+    // port
+    // hostname
+    // inspect
+    // clean: flags["no-incremental-build"],
+  }
+  const {
+    rootFolder,
+    transformFiles,
+    devFolder,
+    ignore,
+    include,
+    isTypescript,
+    writeManifestFile,
+    watch,
+    clean,
+    buildFolder,
+  } = await normalize({ ...serverConfig, env: "dev" })
+
+  if (dev) {
+    // dev
+    const stages = configureStages({ writeManifestFile, isTypescript })
+
+    await transformFiles(rootFolder, stages, devFolder, {
+      ignore,
+      include,
+      watch,
+      clean,
+    })
+  } else {
+    // prod
+    if (!(await alreadyBuilt(buildFolder))) {
+      debug("not built - building..")
+      await build(serverConfig)
+      debug('build done.')
+    } else {
+      debug("already built")
+    }
```

Comment:
I kinda think this would be better in this case - less magic

```suggestion
    if (!(await alreadyBuilt(buildFolder))) {
      throw new Error('Blitz is not built - make sure to building it before running')
    }
```